x-pack/plugins/security/public/management/role_mappings/edit_role_mapping/edit_role_mapping_page.tsx

```diff
@@ -211,6 +214,49 @@ export class EditRoleMappingPage extends Component<Props, State> {
     );
   };
 
+  private checkEmptyAnyAllMappings = (obj: RoleMappingRule) => {
+    const arrToCheck: RoleMappingRule[] = [obj];
+
+    while (arrToCheck.length > 0) {
+      const currentObj = arrToCheck.pop();
+      if (typeof currentObj === 'object' && currentObj !== null) {
```

Comment:
nit
No strong preference, but we may extract it to a function 
`const isObject = (record?: any): record is object => typeof record === 'object' && record !== null`

```suggestion
      if (isObject(currentObj)) {
```

Reply:
Good call - created and used here: https://github.com/elastic/kibana/pull/189340/commits/5473870a1457233e7ba420f688350c1f5c262fb7